### ISSUE TITLE
Suggestion: lint against side-effectful map

### ISSUE THREAD

**Issue by workingjubilee** [rust-lang/rust-clippy#6014]
### What it does

Lint against side-effects in map, or at least println!

### Categories (optional)

- Kind: Correctness or Style

rustc already lints against unused iterators, but sometimes people "fix" this by marking the iterator as discarded. Clippy should be more suspicious: It is not very likely an experienced Rustacean really meant to use `map` and not `for_each` here.

### Drawbacks

Occasionally, sometimes this will be on purpose.

### Example

```rust
fn main() {
    let values = vec!["a", "b", "c"];
    let _results = values.into_iter()
        .map(|val| {
            println!("Value: {}", val);
        });
}
```

Could be written as:

```rust
fn main() {
    let values = vec!["a", "b", "c"];
    let _results = values.into_iter()
        .for_each(|val| {
            println!("Value: {}", val);
        });
}
```

**Comment by ai-mannamalai** [comment]
see also, adjacent, issue https://github.com/rust-lang/rust-clippy/issues/16834
